Perfecto integration documentation in full (#896)
codemagic-ci-cd/codemagic-docs: content/integrations/perfecto-integration.md

---
title: Perfecto integration
description: How to integrate your workflows with Perfecto using codemagic.yaml
weight: 13
---

> **Perfecto** is a web-based Software as a Service (SaaS) platform that allows mobile application developers and QA Engineers to work with services such as advanced automation, monitoring and testing services. It is possible to integrate with Perfecto directly from your **codemagic.yaml**

Reviewer: ```suggestion
**Perfecto** is a web-based platform that allows mobile application developers and QA Engineers to work with services such as advanced automation, monitoring, and testing services. It is possible to integrate with Perfecto directly from your **codemagic.yaml**
```

Signing up with [Perfecto](https://www.perfecto.io/) is required in order to get credentials that are needed during an upload process. 

> Usin the following cURL script in a pre-build script(a script that is run after executing build commands in yaml), **.apk**, **.aab** and **.ipa** binaries can be uploaded to the Perfecto platform:

Reviewer: ```suggestion
Usin gthe following cURL script in a post-build script(a script that is run after executing build commands in yaml), **.apk**, **.aab** and **.ipa** binaries can be uploaded to the Perfecto platform:
```

```
curl "https://web.app.perfectomobile.com/repository/api/v1/artifacts" -H "Perfecto-Authorization: $PERFECTO_TOKEN" -H "Content-Type: multipart/form-data" -F "requestPart={\"artifactLocator\":\"PRIVATE:app.aab\",\"artifactType\":\"ANDROID\",\"override\":true}" -F "inputStream=@/path/to/your_binary"
```

> **PERFECTO_TOKEN** can found in the Perfecto UI with your account. Environment variables can be added in the Codemagic web app using the ‘Environment variables’ tab. You can then and import your variable groups into your codemagic.yaml. For example, if you named your variable group ‘browserstack_credentials’, you would import it as follows:

Reviewer: ```suggestion
**PERFECTO_TOKEN** can be found in the Perfecto UI with your account. Environment variables can be added in the Codemagic web app using the ‘Environment variables tab. You can then import your variable groups into your codemagic.yaml. For example, if you named your variable group ‘perfecto_credentials’, you would import it as follows:
```

            maven {
                url "https://repo1.perfectomobile.com/public/repositories/maven"        
            }
    }
    dependencies {
        classpath "com.perfectomobile.instrumentedtest.gradleplugin:plugin:+"    
    }
}
apply plugin: 'com.perfectomobile.instrumentedtest.gradleplugin'
perfectoGradleSettings {
    configFileLocation "configFile.json"}
```
3. Create **configFile.json** and add the following content in there:
```
{
    "cloudURL": "beta.perfectomobile.com",
    "securityToken":"xxxxxxxxxxxx",
   
	
	"appPath":"repository:PATH_TO_IPA",
	"hostedTestModuleName":"RunnerTests",
	"isHostedTestModule":true,
	
	
	"devices": [
			{"deviceName":"00008020-000D2CC42ED8002E"},
			{"deviceName":"00008101-000B05283081401E"}
	],
	"shard": false,
	"jobName": "Flutter_iOS_Job",
  	"jobNumber": 1,
  	"branch": "Flutter_Branch",
  	"projectName": "My_Flutter_iOS_Project",
  	"projectVersion": "v1.0",
  	"tags": [
    "XCUI", "plugin"  ],
	"takeScreenshotOnTestFailure": false,
	"takeScreenshotOnTestEnd": false,
	"takeScreenshotOnTestStep": false,
	"runUITests":true,
	"runUnitTests":false,
	"installationDetails": {
		"resign": true
	  },
	  
  	"numOfDevices": 2
  }
```

> 4. In order to successfully upload files and enable test automation, the following commands needs to be executed:

Reviewer: ```suggestion
4. In order to successfully upload files and enable test automation, the following commands need to be executed:
```
